compiler/compiler.py

```diff
@@ -122,7 +122,36 @@ def visit_WhileStatement(self, node):
         self.loop_switch_stack.exit_item(node)
         return code
 
-    def visit_BreakStatement(self,node):
+    def visit_SwitchStatement(self, node):
+        self.loop_switch_stack.add_item(node)
+        code = ""
+        end_switch_label = "end_switch_" + str(id(node))
+        for case in node.cases.nodes:
+            end_case_label = "end_case_"+str(id(case))
+            code += self.visit(node.expression)
+            code += self.visit(case.nodes[0])
+            code += "pop eax\n"
+            code += "cmp eax,[esp]\n"
+            code += "add esp,4\n"
+            code += "jne " + end_case_label + "\n"
+            code += self.visit(case.nodes[1])
+            code += end_case_label+":\n"
+
+        if node.default is not None:
+            code += self.visit(node.default)
+
+        code += end_switch_label + ":\n"
+
+        self.loop_switch_stack.exit_item(node)
+
+        return code
+
+    def visit_BreakStatement(self, node):
         top_item = self.loop_switch_stack.get_top_loop_switch()
-        code = "jmp __while_label_end"+str(id(top_item)) + ":\n"
+        if isinstance(top_item, WhileStatement):
+            code = "jmp __while_label_end" + str(id(top_item)) + ":\n"
+        elif isinstance(top_item, SwitchStatement):
+            code = "jmp " + "end_switch_" + str(id(top_item)) + "\n"
+        else:
+            raise Exception("Invalid syntax")
```

Comment:
It would be nice if someone created an `InvalidSyntaxError` sub class of exception,
but that is for another pull request